### DISCUSSION TITLE
Does Arbor support stimulation of cells after they are created?

### DISCUSSION THREAD

**Discussion by lungsi** [arbor-sim/arbor#1885] [Q&A] [answered]
I was wondering if Arbor supports setting up stimulation or stimulation parameters after the cell has already been created? I would like to inject current (`arbor.iclamp`) into a created cell.

All the [tutorial/examples](https://docs.arbor-sim.org/en/stable/tutorial/index.html) shows setting up for stimulation (say, `decor.place('"midpoint"', arbor.iclamp( 10, 2, 0.8), "iclamp")`) as part of setting up the `arbor.decor()` object.

Since the decor object along with the objects for tree and labels are passed as arguments to create the cell (say, `cell = arbor.cable_cell(tree, labels, decor)`), does that mean that one cannot set the stimulation (like `arbor.iclamp`) **after** creating the cell?

**Comment by brenthuisman** [comment, marked as answer]
The API currently assumes that at time of creation of the simulation object, all stimulation has been defined. It follows a functional pattern: the recipe object is passed in, to be examined and the description therein used to construct the simulation. 

You are not the first making the point that this hampers certain kinds of studies, and we're thinking about how to change the API to both keep recipes easy and efficient to probe, and enable more sorts of studies.

I thought we had an issue for this, but I can't find it. If you have the time, you could write that issue, if possible with a clear use case (those are really helpful).
